modules/other/vm_with_power_automate_desktop/main.tf

```diff
@@ -0,0 +1,125 @@
+terraform {
+  required_version = "~> 1.3"
+
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "~> 3.48"
+    }
+    random = {
+      source  = "hashicorp/random"
+      version = "~> 3.5"
+    }
+  }
+
+  backend "azurerm" {}
+}
+
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_public_ip" "public_ip" {
+  count = var.enable_public_access ? 1 : 0
+
+  name                = "public-ip-${var.virtual_machine_name}"
+  location            = var.location
+  resource_group_name = var.resource_group_name
+  allocation_method   = "Static"
+  sku                 = "Standard"
+}
+
+resource "azurerm_network_interface" "network_interface" {
+  name                = "nic-${var.virtual_machine_name}"
+  location            = var.location
+  resource_group_name = var.resource_group_name
+
+  ip_configuration {
+    name                          = "internal"
+    subnet_id                     = var.subnet_id
+    private_ip_address_allocation = "Dynamic"
+    public_ip_address_id          = var.enable_public_access ? azurerm_public_ip.public_ip[0].id : null
+  }
+}
+
+resource "azurerm_network_security_group" "nsg" {
+  count = var.enable_public_access ? 1 : 0
+
+  name                = "nsg-${var.virtual_machine_name}"
+  location            = var.location
+  resource_group_name = var.resource_group_name
+
+  security_rule {
+    name                    = "Allow_RDP"
+    priority                = 100
+    direction               = "Inbound"
+    access                  = "Allow"
+    protocol                = "Tcp"
+    source_port_range       = "*"
+    destination_port_range  = "3389"
+    source_address_prefix   = "Internet"
+    source_address_prefixes = var.allow_rdp_ip
```

Comment:
You removed `destination_address_prefixes` but you are required to have atleast one of `destination_address_prefix` or `destination_address_prefixes`

Reply:
fixed